app/prepare_proposal.go

```diff
@@ -72,8 +68,6 @@ func (app *App) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePr
 		panic(err)
 	}
 
-	// create the new data root by creating the data availability header (merkle
-	// roots of each row and col of the erasure data).
```

Comment:
The following lines don't actually create the data root. That only happens at the very end of the function body with `dah.Hash()`.